coderd/httpmw/provisionerdaemon.go

```diff
@@ -71,16 +71,17 @@ func ExtractProvisionerDaemonAuthenticated(opts ExtractProvisionerAuthConfig) fu
 				return
 			}
 
-			id, keyValue, err := provisionerkey.Parse(key)
+			err := provisionerkey.Validate(key)
 			if err != nil {
-				handleOptional(http.StatusUnauthorized, codersdk.Response{
+				handleOptional(http.StatusBadGateway, codersdk.Response{
```

Comment:
`http.StatusBadGateway`? This should be unauthorized, no?

Reply:
This error has a detail that specifies the length must be 64, and I thought it felt more like a 400 in that regard.

Reply:
omg I meant `StatusBadRequest`...